src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -218,4 +220,31 @@ private void validateClusteringOrder(@Nullable Schema schema) {
       }
     }
   }
+
+  private void validateOperationProperties(@Nullable Schema schema) {
+    if (Arrays.stream(Operation.values()).map(Enum::name).noneMatch(operation.toUpperCase()::equals)) {
+      throw new InvalidConfigPropertyException(String.format("'%s' is incorrect value for field 'Operation'. " +
+                                                               "This field should contain one of the next values: " +
+                                                               "INSERT, UPDATE or UPSERT.", operation), NAME_OPERATION);
+    }
+    if (!Operation.INSERT.equals(getOperation()) && getRelationTableKey() == null) {
+      throw new InvalidConfigPropertyException(
+        "Table key value should be set in case UPDATE or UPSERT operations selected.", NAME_TABLE_KEY);
```

Comment:
Table key must be set if the operation is 'UPDATE' or 'UPSERT'

Reply:
Done.